myoolala/react-starter
# Initial mass updates

## Acceptance Criteria
* Run the app locally with docker for development
* Store session keys in redis
* create an apache container to serve the front end
* Creates the base devops container to setup the repo for terraform
* Removes the need for yarn

## Change Description
Lots of small changes here to get the repo in a better place for a react starter. Namely, adding docker support, begins the terraform process, and adds redis support

## Verification Instructions
* run `docker-compose up --build app` and confirm the app build and runs without issue
* run `docker-compose exec redis redis-cli` and run `KEYS *` to confirm the sessions are being saved into redis
* run `docker-compose build devops && docker-compose run devops` to confirm the devops container builds and runs
* run `docker compose -f docker-compose-httpd.yml up --build httpd` to confirm the app works with a static build running in apache with a node api backend

## Commit Message
BREAKING CHANGE: Adds numerous changes the make the repo more starter friendly
    
    feat: Adds a local docker build option
    feat: Adds a new script to build a static version of the app
    feat: Adds a more production focus node based docker build
    feat: Adds an httpd docker file to allow serving the ui outside of the api container
    docs: Updates readme to account for the new docker build
    feat: Adds a base devops container for terragrunt and terraform work
    feat: Updates yarn to npm
    feat: Adds redis support for remote session storage
    feat: Adds env var to allow for a cleaner method of importing environment based state
    chore: Removes reference to the master branch
